contrib/swagger.yaml

```diff
@@ -0,0 +1,439 @@
+          description: |
+            Forbidden. User tries to update a snippet by another author or anonymous.
+        404:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Not found. Snippet with the given ID value does not exist.
+    delete:
+      summary: Delete the specific code snippet given an id value.
+      description: Delete the specific code snippet given an id value.
+      parameters:
+        - name: id
+          type: integer
+          minimum: 1
+          format: int64
+          required: true
+          in: path
+          description: Unique identifier representing the specific code snippet.
+      responses:
+        200:
+          description: The snippet has been successfully deleted.
+        401:
+          description: |
+            Unauthorized. Only authorized authors are allowed to delete code snippets.
+          schema:
+            $ref: '#/definitions/Error'
+        403:
+          description: |
+            Forbidden. Authors are only allowed to delete snippets they own.
+          schema:
+            $ref: '#/definitions/Error'
+        404:
+          description: The snippet not found.
+          schema:
+            $ref: '#/definitions/Error'
+
+  /snippets/{id}/changesets:
+    get:
+      summary: Retrieve all changesets of the snippet.
+      description: Retrieve all changesets of the snippet.
+      parameters:
+        - name: id
+          type: integer
+          minimum: 1
+          format: int64
+          required: true
+          in: path
+          description: Unique identifier representing the specific code snippet.
+      responses:
+        200:
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Changeset'
+          description: A list of the snippet changesets.
+        400:
+          schema:
+            $ref: '#/definitions/Error'
+          description: Bad request.
+
+  /snippets/{id}/changesets/{revision}:
+    get:
+      summary: Retrieve the specified revision of the snippet.
+      description: Retrieve the specified revision of the snippet.
+      parameters:
+        - name: id
+          type: integer
+          minimum: 1
+          format: int64
+          required: true
+          in: path
+          description: Unique identifier representing the specific code snippet.
+        - name: revision
+          type: integer
+          minimum: -1
+          format: int64
+          required: true
+          in: path
+          description: Changeset index.
+      responses:
+        200:
+          schema:
+            $ref: '#/definitions/Changeset'
+          description: The snippet changeset.
+        400:
+          schema:
+            $ref: '#/definitions/Error'
+          description: Bad request.
+
+definitions:
+  Syntax:
+    type: string
+    enum: *ALLOWED_SYNTAXES
+
+  Changeset:
+    type: object
+    properties:
+      revision:
```

Comment:
Not that I'm against it, but then we should consider adding a ``HEAD`` action for  ``/snippets/{id}/changesets``, so that it's possible to retrieve the current changeset index without downloading all changesets content (what ``GET`` what do).